app/controllers/v1/applications_controller.rb

```diff
@@ -1,8 +1,14 @@
 class V1::ApplicationsController < ApplicationController
   def create
     application = LegalAidApplication.new
+    proceeding_types_found = true
 
-    if application.save
+    if params[:proceeding_type_codes]
+      proceeding_types_found = process_proceeding_types(application)
+      application.proceeding_types << ProceedingType.where(code: params[:proceeding_type_codes])
```

Comment:
Between this line and https://github.com/ministryofjustice/laa-apply-for-legalaid-api/blob/5fdc0d560faa4d43337526dc0e8bdc7b28fb61d1/app/controllers/v1/applications_controller.rb#L25 they're executing exactly the same code, feels like this shouldn't be necessary.

Can't this just be simplified to something like the following? (just a suggestion):

```ruby
  proceeding_types = ProceedingType.where(code: params[:proceeding_type_codes])
  all_match = proceeding_types.size == params[:proceeding_type_codes].size
  application.proceeding_types = proceeding_types if all_match

  if application.save
  ...
  end
```

Reply:
done

Reply:
I realised that RuboCop has issues -
app/controllers/v1/applications_controller.rb:2:3: C: Metrics/AbcSize: Assignment Branch Condition size for create is too high. [18.17/18]